test/evergreen/msan_suppression_wrappers.c

```diff
@@ -0,0 +1,58 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <sys/stat.h>
+#include "string.h"
+
+/*
+ * FIXME-WT-13103 If you add this file to the list of compiled sources and build with the
+ * `-Wl,--wrap=XXX` linker flag the compiler will wrap calls to function XXX with the respective
+ * __wrap_XXX function defined below. We need this because MSan reports false positives when setting
+ * memory via *stat() calls, so we've defined __wrap_*stat functions that first zero the memory to
+ * suppress the false positive. Once WiredTiger's minimum supported LLVM version is 14 this fix
+ * (https://github.com/llvm/llvm-project/commit/4e1a6c07052b466a2a1cd0c3ff150e4e89a6d87a) is
+ * available and this file can be deleted.
+ */
```

Comment:
Should we consider looking at the compiler version and warn if we think if the issue should have been resolved.

Reply:
I was a but wary of doing this since we could end up in a scenario where some of our platforms have LLVM 14 and report the warning while older machines on LLVM <= 13 still require the fix. 
We only have the one MSan variant though and I don't expect that to change; I'll add a warning thanks!

Reply:
Added in [b8d5c5d](https://github.com/wiredtiger/wiredtiger/pull/10685/commits/b8d5c5d51b4b84fb07cd17fad315204dab97ff4f)[b8d5c5d](https://github.com/wiredtiger/wiredtiger/pull/10685/commits/b8d5c5d51b4b84fb07cd17fad315204dab97ff4f)[b8d5c5d](https://github.com/wiredtiger/wiredtiger/pull/10685/commits/b8d5c5d51b4b84fb07cd17fad315204dab97ff4f)